homeassistant/components/romy/config_flow.py

```diff
@@ -0,0 +1,160 @@
+"""Config flow for ROMY integration."""
+from __future__ import annotations
+
+from typing import Any
+
+import romy
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.components import zeroconf
+from homeassistant.const import CONF_HOST, CONF_PASSWORD
+from homeassistant.data_entry_flow import FlowResult
+import homeassistant.helpers.config_validation as cv
+
+from .const import DOMAIN, LOGGER
+
+
+def _schema_with_host() -> vol.Schema:
+    return vol.Schema(
+        {
+            vol.Required(CONF_HOST): cv.string,
+        },
+    )
+
+
+def _schema_with_password() -> vol.Schema:
+    return vol.Schema(
+        {vol.Required(CONF_PASSWORD): vol.All(cv.string, vol.Length(8))},
+    )
+
+
+class RomyConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle config flow for ROMY."""
+
+    VERSION = 1
+    CONNECTION_CLASS = config_entries.CONN_CLASS_LOCAL_POLL
+
+    def __init__(self) -> None:
+        """Handle a config flow for ROMY."""
+        self.host: str = ""
+        self.password: str = ""
+        self.robot_name_given_by_user: str = ""
+
+    async def async_step_user(
+        self, user_input: dict[str, str] | None = None
+    ) -> FlowResult:
+        """Handle the user step."""
+        errors: dict[str, str] = {}
+
+        if user_input:
+            self.host = user_input[CONF_HOST]
+
+            new_romy = await romy.create_romy(self.host, "")
+
+            if not new_romy.is_initialized:
+                errors[CONF_HOST] = "cannot_connect"
+            else:
+                await self.async_set_unique_id(new_romy.unique_id)
+                self._abort_if_unique_id_configured()
+
+                self.robot_name_given_by_user = new_romy.user_name
+
+                if not new_romy.is_unlocked:
+                    return await self.async_step_password()
+
+            if not errors:
+                return await self._async_step_finish_config()
+
+        return self.async_show_form(
+            step_id="user", data_schema=_schema_with_host(), errors=errors
+        )
+
+    async def async_step_password(
+        self, user_input: dict[str, str] | None = None
+    ) -> FlowResult:
+        """Unlock the robots local http interface with password."""
+        errors: dict[str, str] = {}
+
+        if user_input:
+            self.password = user_input[CONF_PASSWORD]
+            new_romy = await romy.create_romy(self.host, self.password)
+
+            if not new_romy.is_initialized:
+                errors[CONF_HOST] = "cannot_connect"
+                return self.async_show_form(
+                    step_id="user", data_schema=_schema_with_host(), errors=errors
```

Comment:
This should stay on the password step and show the password schema again? or is the issue here that the host may have changed? 

It's not allowed to show a form for another step without first moving to that step. i think just showing the password form is simpler here than correcting that...

Another alternative is to just abort.

Reply:
change back to password, problem is if robot gets turned off which is real corner case imho, it never enters the password step again and shows a config flow error:
![image](https://github.com/home-assistant/core/assets/2132412/ff12c3ec-1e97-4dc7-b75e-20244d5c7328)

any idea how can i avoid that? looks like zeroconf kills the flow in the background:
2023-12-19 12:58:12.445 DEBUG (MainThread) [homeassistant.components.zeroconf] service_update: type=_aicu-http._tcp.local. name=aicu-aicgsbksisfapcjqmqjq._aicu-http._tcp.local. state_change=ServiceStateChange.Removed

So abort will also not work cause it never enters async_step_password again

test the case where robot does only not answer http calls so initialized is still false works now like it should and shows up a "Failed to connect" error